app/src/main/res/raw/binarization_shader.glsl

```diff
@@ -3,11 +3,29 @@ precision mediump float;
 
 uniform vec2 resolution;
 uniform sampler2D framebuffer;
+uniform int windowSize;
+uniform float threshold;
 
 void main() {
-    vec2 uv = gl_FragCoord.xy / resolution;
+    float w = 1.0 / resolution.x;
+    float h = 1.0 / resolution.y;
+    vec2 d = vec2(w, h);
 
-    vec4 binarized = step(vec4(0.5), texture2D(framebuffer, uv));
+    vec2 uv = gl_FragCoord.xy / resolution + d/2.0;
```

Comment:
unsure whether we should still use d for this, since i dont remember it making a difference

Reply:
lets just keep it for now